contrib/jms/README.md

```diff
@@ -0,0 +1,45 @@
+# JMS module
+
+This library provides Dataflow sources and sinkgs to make it possible to read
+and write on JMS brokers from Dataflow pipelines.
+
+It supports both JMS queues and topics, with unbounded or bounded `PCollections`.
+
+To use JmsIO, you have to:
+
+1. Create a JMS `ConnectionFactory` specific to the JMS broker you want to use (for instance, ActiveMQ, IBM MQ, ...)
+2. Specify the JMS destination (queue or topic) you want to use
+
+## Reading (consuming messages) with JmsIO
+
+The `JmsIO.Read` transform continuously consumes from the JMS broker and returns an unbounded `PCollection` of `Strings` that
+represent the messages. By default, each element in the resulting `PCollection` is encoded as a UTF-8 string.
+You can override the default encoding by using `withCoder` when you call JmsIO.Read.
+
+----
+PipelineOptions options = PipelineOptionsFactory.create();
+Pipeline p = Pipeline.create(options);
+
+// create JMS connection factory, for instance, with ActiveMQ
+javax.jms.ConnectionFactory connectionFactory = new ActiveMQConnectionFactory("tcp://localhost:61616");
+
+PCollection<String> data = p.apply(JmsIO.Read.named("ConsumeFromJMS").connectionFactory(connectionFactory).queue("my-queue"));
```

Comment:
drop the `.named` code, in favor of just applying the name during `p.apply`. I think its just unnecessary code in this case.